### ISSUE TITLE
Accessing Nested Objects [Not similar #7191]

### ISSUE THREAD

**Issue by Tanvir-rahman** [FreeCodeCamp/FreeCodeCamp#11776]
#### Challenge Name

https://www.freecodecamp.com/challenges/accessing-nested-objects

#### Issue Description

Object's value can be accessed in different ways.I am facing a peculiar problem.If i keep the first answer with comment,then system is accepting my answer **[Image 1]**.  But if i remove the line no 15 **[Image2]** system is not accepting my answer anymore.Though i think those 2 answers are true.

#### Browser Information

* Browser Name, Version:  Google Chrome, Version 54.0.2840.99 m (64-bit)
* Operating System: Windows 10
* Mobile, Desktop, or Tablet: Desktop

Also faced in Firefox, 50.0

#### Your Code

```js
 // Setup
var myStorage = {
  "car": {
    "inside": {
      "glove box": "maps",
      "passenger seat": "crumbs"
     },
    "outside": {
      "trunk": "jack"
    }
  }
};

// Only change code below this line
var gloveBoxContents = myStorage.car["inside"]["glove box"];
 
 
```
#### Screenshot

**Image 1:**
![1](https://cloud.githubusercontent.com/assets/15964566/20561457/edccde88-b1a8-11e6-88a4-8e1f7a1611d7.png)

**Image 2:**
![2](https://cloud.githubusercontent.com/assets/15964566/20561468/f667ed12-b1a8-11e6-8822-609c8271e2d6.PNG)

**Comment by Dayfiri** [comment]
@Tanvir-rahman 
The problem you are having is not a bug and I will confirm it with picture down below but for now try to read the assignment again. The assignment said that:
` Use bracket notation for properties with a space in their name.`
So you need to use bracket notation only on word or words that have have spaces in them. Word `inside` is **not** one of them.
![screenshot from 2016-11-23 18-35-38](https://cloud.githubusercontent.com/assets/16082563/20572525/161f1ac4-b1ac-11e6-9e3a-6b046e06d7ce.png)
This is the working example and you see now what I am talking about. 
Thank you for the report and in the future if you have some questions, the easiest way to receive help is  [here](https://gitter.im/FreeCodeCamp/Help), on a FreeCodeCamp help chat.
Happy coding ! :)

**Comment by Tanvir-rahman** [comment]
I have answered in another way.And system is accepting it.**[Image 1]**..Why?

**Comment by raisedadead** [comment]
@Tanvir-rahman, @Dayfiri 's analogy about the solution is correct. 

However,

TL;DR : COMMENTS in Free Code Camp's editor are not neglected, while testing.

Longer Version:
The challenge does pass with the code :
```
 // Setup
var myStorage = {
  "car": {
    "inside": {
      "glove box": "maps",
      "passenger seat": "crumbs"
     },
    "outside": {
      "trunk": "jack"
    }
  }
};

// Only change code below this line
//var gloveBoxContents = myStorage.car.inside["glove box"];
var gloveBoxContents = myStorage.car["inside"]["glove box"];
```

This is because our challenge infrastructure is run within the browser, meaning all of the code in the code editor is parsed as string, and then executed in a container within the browser's javascript engine.

Unfortunately, this involves a complexity of not being able to parse the commented code 100% accurately.  So yes, this is a bug. But having said that we have had looked at various strategies to have a 100% pure JS parser-interpreter, but it's been confirmed over long discussions that having a full proof solutions led us to more complex infrastructure than it already is.

Remember we currently test all of the code at client side.

Small, inconvenience in this case is do not add code in comments which you do not need, you are free to have text comments though. Or should you need to keep alternate solutions in the comments, re-name their variable names, than those used in the seed.

Thanks and happy coding. Apologies but will be closing as a wontfix.
